src/pages/Profile.jsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import { AuthContext } from '../context/AuthContext';
+import { useContext } from 'react';
+import { useState, useEffect } from 'react';
+import axios from 'axios';
+
+function Profile() {
+
+  const { email, username } = useContext(AuthContext);
+  const [ privateData, setPrivateData ] = useState({});
```

Comment:
Ik zou de default waarde `undefined` maken, zodat je een beter (te controleren) onderscheid hebt tussen of het al wel of niet is ingeladen
```suggestion
  const [ privateData, setPrivateData ] = useState();
```